validator/sawtooth_validator/execution/scheduler.py

```diff
@@ -101,9 +101,18 @@ def finalize(self):
         raise NotImplementedError()
 
     @abstractmethod
-    def complete(self, block):
+    def complete(self, block, force=False):
```

Comment:
How about force_persist? Because there would be other times that the tree would be persisted (Like in block validation where a resulting state root is given and it matches what transaction execution finds to be the state root -- it will persist that).